Add cluster awareness and decommission docs (#2438)
opensearch-project/documentation-website: _ml-commons-plugin/cluster-settings.md

## Run tasks and models on ML nodes only

> If `true`, ML Commons tasks and models run machine learning (ML) tasks on ML nodes only. If `false`, tasks and models run on ML nodes first. If no ML nodes exist, tasks and models run on data nodes. Don't set as "false" on production cluster. 

Reviewer: don't set --> We recommend that you do not set this value to "false".... ?

> If `true`, ML Commons tasks and models run machine learning (ML) tasks on ML nodes only. If `false`, tasks and models run on ML nodes first. If no ML nodes exist, tasks and models run on data nodes. Don't set as "false" on production cluster. 

Reviewer 2: ```suggestion
If `true`, ML Commons tasks and models run machine learning (ML) tasks on ML nodes only. If `false`, tasks and models run on ML nodes first. If no ML nodes exist, tasks and models run on data nodes. We recommend that you do not set this value to "false" on production clusters. 
```

### Setting

```
plugins.ml_commons.only_run_on_ml_node: true
```

### Values

- Default value: `false`
- Value range: `true` or `false`

## Dispatch tasks to ML node 

> `round_robin` dispatches ML tasks to ML nodes using round robin routing. `least_load` gathers all an ML nodes' runtime information, like JVM heap memory usage and running tasks, then dispatches tasks to the ML node with the least load.

Reviewer: all an ML --> all ML
```suggestion
`round_robin` dispatches ML tasks to ML nodes using round robin routing. `least_load` gathers runtime information from all ML nodes, like JVM heap memory usage and running tasks, then dispatches the tasks to the ML node with the least load.
```

Reviewer 2: Should the last two words be "lowest load"? low - lower - lowest; little - less/lesser - least

> ## Set sync up job intervals 

Reviewer 2: Global: Do we have to use "sync-up job" (note the hyphen), or can we call it something else? "Sync-up job" is awkward/clunky.

Author: Does "sync job" sound better?

Reviewer 2: Yes

## Load model tasks per node

> Controls how many load model tasks can run in parallel on one node. If set as 0, you cannot load models to any node.

Reviewer: as --> to (set to 0)

> Controls how many load model tasks can run in parallel on one node. If set as 0, you cannot load models to any node.

Reviewer 2: ```suggestion
Controls how many load model tasks can run in parallel on one node. If set to 0, you cannot load models to any node.
```

### Setting

```
plugins.ml_commons.max_load_model_tasks_per_node: 10
```

### Values 

- Default value: `10`
- Value range: [0, 10]

## Add trusted URL

> The default value allows uploading a model file from any http/https/ftp/local file. You can change this value to restrict trusted model URL.

Reviewer: restrict trusted model URLs.  (not sure if it's supposed to be singular or plural)

> The default value allows uploading a model file from any http/https/ftp/local file. You can change this value to restrict trusted model URL.

Reviewer 2: ```suggestion
The default value allows you to upload a model file from any http/https/ftp/local file. You can change this value to restrict trusted model URLs.
```

Reviewer 2: Should "HTTP/HTTPS/FTP" be in all caps?

### Setting

```
plugins.ml_commons.trusted_url_regex: ^(https?\|ftp\|file)://[-a-zA-Z0-9+&@#/%?=~_\|!:,.;]*[-a-zA-Z0-9+&@#/%=~_\|]
```

### Values

- Default value: `^(https?\|ftp\|file)://[-a-zA-Z0-9+&@#/%?=~_\|!:,.;]*[-a-zA-Z0-9+&@#/%=~_\|]`
- Value range: Java regular expression (regex) string

## Assign task timeout

> Assigns how long an ML task will live in seconds. After timeout, the task will fail.

Reviewer: ```suggestion
Assigns how long in seconds an ML task will live. After the timeout, the task will fail.
```

### Setting

```
plugins.ml_commons.ml_task_timeout_in_seconds: 600
```

### Values

- Default value: 600
- Value range: [1, 86400]

## Set native memory threshold 

> Sets a circuit breaker that checks all system memory usage before running an ML task. If the native memory exceeds the threshold, OpenSearch puts up an exception and stops running any ML task. 

Reviewer 2: ```suggestion
Sets a circuit breaker that checks all system memory usage before running an ML task. If the native memory exceeds the threshold, OpenSearch throws an exception and stops running any ML task. 
```

Reviewer 2: I think we mean "throws" here, but if not, please replace the phrasal verb "puts up" with something more descriptive.
